### PR TITLE
Feature/accordion component

### PR DESCRIPTION
**Pull request by AliShirzad**

### CODE REVIEW

File: libs/components/src/lib/Accordion/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,88 @@
+@import "../../../../../libs/theme";
+
+.container {
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+
+  &__item {
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: use unique class names for all items

File: libs/components/src/lib/Accordion/index.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { PlusIcon } from "../Icons/plus";
+import classes from "./index.module.scss";
+
+interface AccordionItemProps {
+  title: string;
+  children: ReactNode;
+}
+
+const AccordionItem: FC<AccordionItemProps> = ({ title, children }) => {
+  const [isOpen, setIsOpen] = useState(false);
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.tsx]: index file must be contained just jsx codes, move state or functions to index.hook file

File: libs/components/src/lib/Accordion/index.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          className={cls(classes.container__item__header__title, {
+            [classes.container__item__header__title__open]: isOpen,
+          })}
+          role="contentinfo"
+        >
+          {title}
+        </span>
+        <div
+          className={cls(classes.container__item__header__icon, {
+            [classes.container__item__header__icon__open]: isOpen,
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.tsx]: use recognizable class names

File: libs/components/src/lib/Accordion/index.module.scss [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+      overflow: hidden;
+      color: $dark;
+      font-family: Montserrat;
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 400;
+
+      &__open {
+        max-height: 1000px; /* or any other large value */
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: remove comment, 1000px ??

File: libs/components/src/lib/Accordion/index.module.scss [hunk 5]
```diff
@@ -0,0 +1,88 @@
+@import "../../../../../libs/theme";
+
+.container {
+  width: 100%;
+  display: flex;
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: we have a mixin for flex (flexbox).
check this file:
/libs/theme/_mixins.scss

File: libs/components/src/lib/Accordion/index.module.scss (hunk 1 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.module.scss]: change container to accordion and item to accordion__item

File: libs/components/src/lib/Accordion/index.tsx (hunk 2 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.tsx]: move useState to new index.hook.ts file

File: libs/components/src/lib/Accordion/index.tsx (hunk 3 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.tsx]: change "container__item__header__icon" to "accordion__item__header__icon" to be more clear

File: libs/components/src/lib/Accordion/index.module.scss (hunk 4 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.module.scss]: removed comment and reduced max-height from 1000 to 800px

File: libs/components/src/lib/Accordion/index.module.scss (hunk 5 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.module.scss]: added @include flexbox to every place that is needed in scss file

File: libs/components/src/lib/Accordion/index.module.scss (hunk 1 above)
Comment (amirmohammad-mohamadi) [reply libs/components/src/lib/Accordion/index.module.scss]: I mean in scss file it is not clear to define classes like this, for example for accordion__item, it's better use like this:
.accordion__item {...} not this: &__item {...}, please modify this for all classes in scss file

File: libs/components/src/lib/Accordion/index.module.scss [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    &__title {
+      width: 100%;
+      @include flexbox(row, center, center);
+      cursor: pointer;
+    }
+
+    &__children {
+      width: 100%;
+      font-size: 16px;
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: for font-size please use like this @include typography, you can see all typography in figma or this path: libs/theme/_mixins.scss
please modify this for all font-sizes

File: libs/components/src/lib/Accordion/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      font-family: Montserrat;
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 400;
+
+      &__open {
+        max-height: 800px;
+        overflow: auto;
+        opacity: 1;
+        padding-bottom: 24px;
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: use spacing

File: libs/components/src/lib/Accordion/index.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import { PlusIcon } from "../Icons/plus";
+import classes from "./index.module.scss";
+import { useData } from "./index.hook";
+
+interface AccordionItemProps {
+  title: string;
+  children: ReactNode;
+}
+
+const AccordionItem: FC<AccordionItemProps> = ({ title, children }) => {
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.tsx]: please use shadow rendering for AccordionItem, In each index file we should have just one export.

File: libs/components/src/lib/Accordion/index.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      </div>
+    </div>
+  );
+};
+
+interface AccordionProps {
+  items?: AccordionItemProps[];
+}
+
+const Accordion: FC<AccordionProps> = ({ items }) => {
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.tsx]: const Accordion to export const Accordion and remove export in line 72.

File: libs/components/src/lib/Accordion/index.module.scss [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+    &:last-child {
+      border: none;
+    }
+
+    &__header {
+      width: 100%;
+      @include flexbox(row, space-between, center);
+
+      padding: 24px 0 18px 0;
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: use spacing

File: libs/components/src/lib/Accordion/index.module.scss [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+      cursor: pointer;
+
+      &__title {
+        font-size: 20px;
```
Comment (amirmohammad-mohamadi) [inline libs/components/src/lib/Accordion/index.module.scss]: use typography

File: libs/components/src/lib/Accordion/index.tsx (hunk 9 above)
Comment (AliShirzad) [reply libs/components/src/lib/Accordion/index.tsx]: ### This needs more discussion. 
AccordionItem manages its own "state". Using a render method like renderAccordion item brings the state to the parent. Then for accessing "state" in renderAccordionItem we only will be able to access "state" like this:

```
const [state, useState] = useState([])
items.map((item, index) => renderAccordionItem(item[index])
```
Managing "state" in the AccordionItem component lowers unnecessary rerenders and any possible implications that using the index in the map function will bring. 

I wanted to request to accept more than one component in one module as standard.
This Reddit post better explained what I am saying.

[Reddit](https://www.reddit.com/r/reactjs/comments/uvhi7n/is_it_good_practice_to_put_multiple_components_in/): __is it good practice to put multiple components in one file like this?_

_I think that is absolutely fine. Think of it as a way of organizing the main component better by defining subcomponents in the same file; there's no benefit in adding cognitive overhead by moving these subcomponents to other files. I personally prefer a slightly larger file with a well-organized structure than having to jump across several files for components that are not being reused._

**Pull request merged**